crates/holochain/src/conductor/chc/chc_remote.rs

```diff
@@ -110,18 +102,111 @@ impl ChcRemoteClient {
         self.base_url.join(path).expect("invalid URL").to_string()
     }
 
-    async fn post(&self, path: &str, body: Vec<u8>) -> ChcResult<reqwest::Response> {
+    async fn post<T>(&self, path: &str, body: &T) -> ChcResult<reqwest::Response>
+    where
+        T: serde::Serialize + std::fmt::Debug,
+    {
         let client = reqwest::Client::new();
         let url = self.url(path);
-        client
-            .post(url)
+        let body = holochain_serialized_bytes::encode(body)?;
+        let res: reqwest::Response = client
+            .post(url.clone())
             .body(body)
             .send()
             .await
-            .map_err(extract_string)
+            .map_err(extract_string)?;
+        Ok(res)
     }
 }
 
 fn extract_string(e: reqwest::Error) -> ChcError {
     ChcError::ServiceUnreachable(e.to_string())
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::test_utils::valid_arbitrary_chain;
+    use pretty_assertions::assert_eq;
+
+    #[tokio::test(flavor = "multi_thread")]
+    #[ignore = "this test requires a remote service, so it should only be run manually"]
```

Comment:
How complex is the remote service? Would it be worth building a stub that we can use in Holochain so that this test can run?

Reply:
As another possibility, you could do what we're doing on the holo end: including the actual service as a git submodule and running it locally for integration tests.

Reply:
@ThetaSinner the chc_local module uses such a stub. This ignored test is explicitly for testing an actual remote service, i.e. integration test with the remote service